src/decision/src/subroutines/bottom_line/FindLine.cpp

```diff
@@ -0,0 +1,34 @@
+//
+// Created by drdc-s2632 on 17/07/18.
+//
+
+#include "FindLine.h"
+
+std::vector<ros::Subscriber> FindLine::getSubscriptions(ros::NodeHandle nh) {
+    std::string gateDetectTopic = "line_detect_output";
+
+    timer_ =
+    nh.createTimer(ros::Duration(10), &FindLine::timerCallback, this, false);
+    timer_.start();
+
+    lateralVelocityDirection_ = 1;
+
+    std::vector<ros::Subscriber> subs;
+
+    subs.push_back(
+    nh.subscribe(gateDetectTopic, 10, &FindLine::lineDetectCallback, this));
+    return subs;
+}
+
+void FindLine::lineDetectCallback(
+const line_detect::LineDetectMsg::ConstPtr& msg) {
+    geometry_msgs::TwistStamped command;
+
+    command.twist.linear.y = lateralVelocityDirection_ * RIGHT;
+
+    publishCommand(command);
+}
+
+void FindLine::timerCallback(const ros::TimerEvent& event) {
```

Comment:
I don't find this super intuitive, why do we reverse this on a timer?

Reply:
Just going right and left to find the line if we don't end up right over it

Reply:
Yeah I didn't have much time to leave a lot of comments on this, sorry about that

Reply:
We def. should have comments in places like this, otherwise this will be impossible to debug if you're not immediately available. In general I would encourage taking the time to put comments and do things cleanly, not much point rushing if we end up with something we can't work with and quickly debug at competition.

Reply:
fixed